### ISSUE TITLE
builtin_op_importers.cpp:5404 In function importFallbackPluginImporter

### ISSUE THREAD

**Issue by Skier23** [NVIDIA/TensorRT#3734]
## Environment

Windows 10

**TensorRT Version**: 8.6.1

**NVIDIA GPU**: 3090

**NVIDIA Driver Version**: Latest

**CUDA Version**: 12.1

**CUDNN Version**:

Operating System: Windows 10

Python Version (if applicable): 3.11

Tensorflow Version (if applicable):

PyTorch Version (if applicable):

Baremetal or Container (if so, version):

## Relevant Files
https://huggingface.co/timm/maxvit_xlarge_tf_384.in21k_ft_in1k

## Steps To Reproduce
trtexec --onnx=out_onnx_model.onnx --saveEngine=engine_pytorch.trt  --explicitBatch

log:

`
&&&& RUNNING TensorRT.trtexec [TensorRT v8601] # C:\Program Files\TensorRT-8.6.1.6.Windows10.x86_64.cuda-12.0\TensorRT-8.6.1.6\bin\trtexec.exe --onnx=out_onnx_model.onnx --saveEngine=resnet_engine_pytorch.trt --explicitBatch
[03/25/2024-09:34:23] [W] --explicitBatch flag has been deprecated and has no effect!
[03/25/2024-09:34:23] [W] Explicit batch dim is automatically enabled if input model is ONNX or if dynamic shapes are provided when the engine is built.
[03/25/2024-09:34:23] [I] === Model Options ===
[03/25/2024-09:34:23] [I] Format: ONNX
[03/25/2024-09:34:23] [I] Model: out_onnx_model.onnx
[03/25/2024-09:34:23] [I] Output:
[03/25/2024-09:34:23] [I] === Build Options ===
[03/25/2024-09:34:23] [I] Max batch: explicit batch
[03/25/2024-09:34:23] [I] Memory Pools: workspace: default, dlaSRAM: default, dlaLocalDRAM: default, dlaGlobalDRAM: default
[03/25/2024-09:34:23] [I] minTiming: 1
[03/25/2024-09:34:23] [I] avgTiming: 8
[03/25/2024-09:34:23] [I] Precision: FP32
[03/25/2024-09:34:23] [I] LayerPrecisions:
[03/25/2024-09:34:23] [I] Layer Device Types:
[03/25/2024-09:34:23] [I] Calibration:
[03/25/2024-09:34:23] [I] Refit: Disabled
[03/25/2024-09:34:23] [I] Version Compatible: Disabled
[03/25/2024-09:34:23] [I] TensorRT runtime: full
[03/25/2024-09:34:23] [I] Lean DLL Path:
[03/25/2024-09:34:23] [I] Tempfile Controls: { in_memory: allow, temporary: allow }
[03/25/2024-09:34:23] [I] Exclude Lean Runtime: Disabled
[03/25/2024-09:34:23] [I] Sparsity: Disabled
[03/25/2024-09:34:23] [I] Safe mode: Disabled
[03/25/2024-09:34:23] [I] Build DLA standalone loadable: Disabled
[03/25/2024-09:34:23] [I] Allow GPU fallback for DLA: Disabled
[03/25/2024-09:34:23] [I] DirectIO mode: Disabled
[03/25/2024-09:34:23] [I] Restricted mode: Disabled
[03/25/2024-09:34:23] [I] Skip inference: Disabled
[03/25/2024-09:34:23] [I] Save engine: resnet_engine_pytorch.trt
[03/25/2024-09:34:23] [I] Load engine:
[03/25/2024-09:34:23] [I] Profiling verbosity: 0
[03/25/2024-09:34:23] [I] Tactic sources: Using default tactic sources
[03/25/2024-09:34:23] [I] timingCacheMode: local
[03/25/2024-09:34:23] [I] timingCacheFile:
[03/25/2024-09:34:23] [I] Heuristic: Disabled
[03/25/2024-09:34:23] [I] Preview Features: Use default preview flags.
[03/25/2024-09:34:23] [I] MaxAuxStreams: -1
[03/25/2024-09:34:23] [I] BuilderOptimizationLevel: -1
[03/25/2024-09:34:23] [I] Input(s)s format: fp32:CHW
[03/25/2024-09:34:23] [I] Output(s)s format: fp32:CHW
[03/25/2024-09:34:23] [I] Input build shapes: model
[03/25/2024-09:34:23] [I] Input calibration shapes: model
[03/25/2024-09:34:23] [I] === System Options ===
[03/25/2024-09:34:23] [I] Device: 0
[03/25/2024-09:34:23] [I] DLACore:
[03/25/2024-09:34:23] [I] Plugins:
[03/25/2024-09:34:23] [I] setPluginsToSerialize:
[03/25/2024-09:34:23] [I] dynamicPlugins:
[03/25/2024-09:34:23] [I] ignoreParsedPluginLibs: 0
[03/25/2024-09:34:23] [I]
[03/25/2024-09:34:23] [I] === Inference Options ===
[03/25/2024-09:34:23] [I] Batch: Explicit
[03/25/2024-09:34:23] [I] Input inference shapes: model
[03/25/2024-09:34:23] [I] Iterations: 10
[03/25/2024-09:34:23] [I] Duration: 3s (+ 200ms warm up)
[03/25/2024-09:34:23] [I] Sleep time: 0ms
[03/25/2024-09:34:23] [I] Idle time: 0ms
[03/25/2024-09:34:23] [I] Inference Streams: 1
[03/25/2024-09:34:23] [I] ExposeDMA: Disabled
[03/25/2024-09:34:23] [I] Data transfers: Enabled
[03/25/2024-09:34:23] [I] Spin-wait: Disabled
[03/25/2024-09:34:23] [I] Multithreading: Disabled
[03/25/2024-09:34:23] [I] CUDA Graph: Disabled
[03/25/2024-09:34:23] [I] Separate profiling: Disabled
[03/25/2024-09:34:23] [I] Time Deserialize: Disabled
[03/25/2024-09:34:23] [I] Time Refit: Disabled
[03/25/2024-09:34:23] [I] NVTX verbosity: 0
[03/25/2024-09:34:23] [I] Persistent Cache Ratio: 0
[03/25/2024-09:34:23] [I] Inputs:
[03/25/2024-09:34:23] [I] === Reporting Options ===
[03/25/2024-09:34:23] [I] Verbose: Disabled
[03/25/2024-09:34:23] [I] Averages: 10 inferences
[03/25/2024-09:34:23] [I] Percentiles: 90,95,99
[03/25/2024-09:34:23] [I] Dump refittable layers:Disabled
[03/25/2024-09:34:23] [I] Dump output: Disabled
[03/25/2024-09:34:23] [I] Profile: Disabled
[03/25/2024-09:34:23] [I] Export timing to JSON file:
[03/25/2024-09:34:23] [I] Export output to JSON file:
[03/25/2024-09:34:23] [I] Export profile to JSON file:
[03/25/2024-09:34:23] [I]
[03/25/2024-09:34:23] [I] === Device Information ===
[03/25/2024-09:34:23] [I] Selected Device: NVIDIA GeForce RTX 3090
[03/25/2024-09:34:23] [I] Compute Capability: 8.6
[03/25/2024-09:34:23] [I] SMs: 82
[03/25/2024-09:34:23] [I] Device Global Memory: 24575 MiB
[03/25/2024-09:34:23] [I] Shared Memory per SM: 100 KiB
[03/25/2024-09:34:23] [I] Memory Bus Width: 384 bits (ECC disabled)
[03/25/2024-09:34:23] [I] Application Compute Clock Rate: 1.695 GHz
[03/25/2024-09:34:23] [I] Application Memory Clock Rate: 9.751 GHz
[03/25/2024-09:34:23] [I]
[03/25/2024-09:34:23] [I] Note: The application clock rates do not reflect the actual clock rates that the GPU is currently running at.
[03/25/2024-09:34:23] [I]
[03/25/2024-09:34:23] [I] TensorRT version: 8.6.1
[03/25/2024-09:34:23] [I] Loading standard plugins
[03/25/2024-09:34:24] [I] [TRT] [MemUsageChange] Init CUDA: CPU +3, GPU +0, now: CPU 38775, GPU 1269 (MiB)
[03/25/2024-09:34:29] [I] [TRT] [MemUsageChange] Init builder kernel library: CPU +1472, GPU +266, now: CPU 41425, GPU 1535 (MiB)
[03/25/2024-09:34:29] [I] Start parsing network model.
[libprotobuf WARNING **************************************************************************\externals\protobuf\3.0.0\src\google\protobuf\io\coded_stream.cc:604] Reading dangerously large protocol message.  If the message turns out to be larger than 2147483647 bytes, parsing will be halted for security reasons.  To increase the limit (or to disable these warnings), see CodedInputStream::SetTotalBytesLimit() in google/protobuf/io/coded_stream.h.
[libprotobuf WARNING **************************************************************************\externals\protobuf\3.0.0\src\google\protobuf\io\coded_stream.cc:81] The total number of bytes read was 948423101
[03/25/2024-09:34:30] [I] [TRT] ----------------------------------------------------------------
[03/25/2024-09:34:30] [I] [TRT] Input filename:   out_onnx_model.onnx
[03/25/2024-09:34:30] [I] [TRT] ONNX IR version:  0.0.8
[03/25/2024-09:34:30] [I] [TRT] Opset version:    1
[03/25/2024-09:34:30] [I] [TRT] Producer name:    pytorch
[03/25/2024-09:34:30] [I] [TRT] Producer version: 2.2.1
[03/25/2024-09:34:30] [I] [TRT] Domain:
[03/25/2024-09:34:30] [I] [TRT] Model version:    0
[03/25/2024-09:34:30] [I] [TRT] Doc string:
[03/25/2024-09:34:30] [I] [TRT] ----------------------------------------------------------------
[libprotobuf WARNING **************************************************************************\externals\protobuf\3.0.0\src\google\protobuf\io\coded_stream.cc:604] Reading dangerously large protocol message.  If the message turns out to be larger than 2147483647 bytes, parsing will be halted for security reasons.  To increase the limit (or to disable these warnings), see CodedInputStream::SetTotalBytesLimit() in google/protobuf/io/coded_stream.h.
[libprotobuf WARNING **************************************************************************\externals\protobuf\3.0.0\src\google\protobuf\io\coded_stream.cc:81] The total number of bytes read was 948423101
[03/25/2024-09:34:30] [I] [TRT] No importer registered for op: timm_models_maxxvit_MaxxVit_base_model_1. Attempting to import as plugin.
[03/25/2024-09:34:30] [I] [TRT] Searching for plugin: timm_models_maxxvit_MaxxVit_base_model_1, plugin_version: 1, plugin_namespace:
[03/25/2024-09:34:30] [E] [TRT] 3: getPluginCreator could not find plugin: timm_models_maxxvit_MaxxVit_base_model_1 version: 1
[03/25/2024-09:34:30] [E] [TRT] ModelImporter.cpp:771: While parsing node number 0 [timm_models_maxxvit_MaxxVit_base_model_1 -> "base_model_1"]:
[03/25/2024-09:34:30] [E] [TRT] ModelImporter.cpp:772: --- Begin node ---
[03/25/2024-09:34:30] [E] [TRT] ModelImporter.cpp:773: input: "l_x_"
...
input: "base_model.stages.3.blocks.1.attn_grid.mlp.fc2.weight"
input: "base_model.stages.3.blocks.1.attn_grid.mlp.fc2.bias"
input: "base_model.head.norm.weight"
input: "base_model.head.norm.bias"
input: "base_model.head.pre_logits.fc.weight"
input: "base_model.head.pre_logits.fc.bias"
output: "base_model_1"
name: "timm_models_maxxvit_MaxxVit_base_model_1_0"
op_type: "timm_models_maxxvit_MaxxVit_base_model_1"
doc_string: ""
domain: "pkg.timm.0.9.16"

[03/25/2024-09:34:30] [E] [TRT] ModelImporter.cpp:774: --- End node ---
[03/25/2024-09:34:30] [E] [TRT] ModelImporter.cpp:777: ERROR: builtin_op_importers.cpp:5404 In function importFallbackPluginImporter:
[8] Assertion failed: creator && "Plugin not found, are the plugin name, version, and namespace correct?"
[03/25/2024-09:34:30] [E] Failed to parse onnx file
[03/25/2024-09:34:30] [I] Finished parsing network model. Parse time: 1.26444
[03/25/2024-09:34:30] [E] Parsing model failed
[03/25/2024-09:34:30] [E] Failed to create engine from model or file.
[03/25/2024-09:34:30] [E] Engine set up failed`

**Comment by zerollzeng** [comment]
> [03/25/2024-09:34:30] [I] [TRT] No importer registered for op: timm_models_maxxvit_MaxxVit_base_model_1. Attempting to import as plugin.

You need to implement a TRT plugin for this custom op.

**Comment by zerollzeng** [comment]
TensorRT's support operators: https://github.com/onnx/onnx-tensorrt/blob/main/docs/operators.md
and we have some prebuilt plugin, for other non-standard operators, user need to implement their own plugin.

**Comment by zerollzeng** [comment]
see https://docs.nvidia.com/deeplearning/tensorrt/developer-guide/index.html#extending

**Comment by ttyio** [comment]
closing since no activity for more than 3 weeks per our policy. thanks!
